src/components/molecules/Common/AssetModal/VirtualAssetCard/index.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useCallback, useState, useEffect } from "react";
+import { styled } from "@reearth/theme";
+
+type CardSize = "small" | "medium" | "large" | "list";
+
+export type Props = {
+  assetsLength?: number;
+  containerWidth?: number;
+  cardSize?: CardSize;
+};
+
+const VirtualAssetCard: React.FC<Props> = ({ assetsLength, containerWidth, cardSize }) => {
+  const [virtualWidth, setVirtualWidth] = useState(0);
+
+  const handleVirtualWidth = useCallback(() => {
+    if (containerWidth) {
+      if (assetsLength) {
+        const cardWidth = cardSize === "medium" ? 216 : 128;
+        const numInrow = Math.floor(containerWidth / cardWidth);
+        const lastRowNum = assetsLength % numInrow;
+        const gapWidth = (containerWidth - cardWidth * numInrow) / (numInrow - 1);
+        const virtualBoxNumber = numInrow - lastRowNum;
+        const virtualBoxWidth = virtualBoxNumber * cardWidth + gapWidth * (virtualBoxNumber - 1);
+        if (lastRowNum === 0) {
+          setVirtualWidth(0);
+        } else {
+          setVirtualWidth(virtualBoxWidth);
+        }
+      }
+    }
```

Comment:
If you can explain in the comments why you want to do this kind of calculation, the code will be understandable for others to read. 

Also, have you considered whether this can be solved with CSS?

Reply:
So you suggest to write some Code remarks here ? sure
I want to create a virtual object which takes the space inside the flex box, which can align the other objects.
yes, I have considered the css way, but I think this is the best way to sovle this problem, any suggetions ?

Reply:
What features do you need it for?
Maybe we can use CSS Grid or something?

Reply:
thanks, I will switch to css grid